site/src/pages/tutorials/thrift/blog/implement-delete.mdx

```diff
@@ -0,0 +1,196 @@
+---
+menuTitle: "Implement DELETE"
+order: 7
+category: thrift
+tags:
+  - server
+level: basic
+type: step
+---
+
+# Implementing DELETE operation
+
+So far, we created, read, and updated a blog post.
+Now, let's implement and make a call to delete a blog post.
+Also, we'll add Armeria's [Documentation Service](/docs/server-docservice) for testing our blog service.
+
+<TutorialSteps current={7} />
+
+## What you need
+
+You need to have the following obtained from previous steps:
+
+- [Generated Java code](/tutorials/thrift/blog/create-service#2-compile-the-thrift-file)
+- Main.java
+- BlogServiceImpl.java
+- BlogServiceExceptionHandler.java
+- BlogClient.java
+- BlogServiceTest.java
+
+## 1. Implement server-side
+
+In the `BlogServiceImpl` class, implement the `deleteBlogPost()` method to delete a blog post.
+Let's throw an exception in case there is no blog post for the given ID.
+
+```java filename=BlogServiceImpl.java
+@Override
+public void deleteBlogPost(DeleteBlogPostRequest request, AsyncMethodCallback<Void> resultHandler)
+      throws TException {
+  final BlogPost removed = blogPosts.remove(request.getId());
+  if (removed == null) {
+    resultHandler.onError(
+            new IllegalArgumentException("The blog post does not exist. ID: " + request.getId()));
+  } else {
+    resultHandler.onComplete(null);
+  }
+}
+```
+
+## 2. Implement client-side
+
+In the `BlogClient` class, add the `deleteBlogPost()` method to send a request deleting a blog post.
+
+```java filename=BlogClient.java
+import example.armeria.blog.thrift.DeleteBlogPostRequest;
+...
+void deleteBlogPost(int id) throws TException {
+  final DeleteBlogPostRequest request = new DeleteBlogPostRequest().setId(id);
+  blogService.deleteBlogPost(request);
+}
+```
+
+## 3. Test deleting a blog post
+
+Let's test deleting a blog post.
+We'll delete the blog post with ID `1`, and try retrieving with the same ID to verify it is indeed deleted.
+Add a test method like the following.
+
+```java filename=BlogServiceTest.java
+@Test
+@Order(7)
+void deleteBlogPost() throws TException {
+  final BlogClient client = new BlogClient(server.httpUri(), "/thrift");
+  client.deleteBlogPost(1);
+  final Throwable exception = catchThrowable(() -> {
+    client.getBlogPost(1);
+  });
+  assertThat(exception).isInstanceOf(BlogNotFoundException.class)
+          .extracting("reason")
+          .asString()
+          .isEqualTo("The blog post does not exist. ID: 1");
```

Comment:
What do you think about breaking line right before `.isInstanceOf()`? We could fix all similar sentences for aesthetics' sake:
```suggestion
  assertThat(exception)
    .isInstanceOf(BlogNotFoundException.class)
    .extracting("reason")
    .asString()
    .isEqualTo("The blog post does not exist. ID: 1");
```
Please note that I also reduced the indentation at minimum level so the code is readable even on mobile devices.